cmd/local-locker.go

```diff
@@ -52,9 +61,25 @@ func isWriteLock(lri []lockRequesterInfo) bool {
 //
 //msgp:ignore localLocker
 type localLocker struct {
-	mutex   sync.Mutex
-	lockMap map[string][]lockRequesterInfo
-	lockUID map[string]string // UUID -> resource map.
+	mutex     sync.Mutex
+	waitMutex atomic.Int32
+	lockMap   map[string][]lockRequesterInfo
```

Comment:
This would be a larger change, but I wonder if this can be a map of `map[string][]lockRequesterInfo` where `lockRequesterInfo` would also have a counter inside which can be incremented and decremented. This way `removeEntry` would not need to remove an item from a slice, but just decrease an integer.

Reply:
Then there will be more to clean up.

We are not going to make further optimizations here and we probably will not take changes that risk destabilizing the setup. Refactoring if out of the scope of this project.